implementation/src/main/java/io/smallrye/metrics/ExtendedMetadata.java

```diff
@@ -26,21 +27,17 @@
 /**
  * @author hrupp
  */
-public class ExtendedMetadata extends Metadata {
+public class ExtendedMetadata extends DefaultMetadata {
```

Comment:
If `org.eclipse.microprofile.metrics.DefaultMetadata` is immutable I'd say our extension should be immutable too, so the setters should rather create a clone while changing one field and return that clone. 

In general, for immutable classes containing larger number of fields, I would consider implementing a Builder, but that would be needed also in `DefaultMetadata`... but then our Builder could extend the Builder of `DefaultMetadata`.

A nice example of Builder inheritance is in the Creaper library for WildFly management: https://github.com/wildfly-extras/creaper/blob/master/commands/src/main/java/org/wildfly/extras/creaper/commands/infinispan/cache/AddReplicatedCache.java which extends https://github.com/wildfly-extras/creaper/blob/master/commands/src/main/java/org/wildfly/extras/creaper/commands/infinispan/cache/AbstractAddCache.java

But that would require creating a builder in `DefaultMetadata`, if we don't want to do these changes on spec level, then we might at least consider what I mentioned earlier, that is creating clones to obtain immutability for `ExtendedMetadata`. Or we could create a builder specifically for `ExtendedMetadata` which calls the constructor of `DefaultMetadata` during building.